### ISSUE TITLE
Build of gitb-reports fails for missing dependency

### ISSUE THREAD

**Issue by longama** [ISAITB/gitb#21]
Hi,

during the maven build I got the following error:

`[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< com.gitb:gitb-reports >------------------------
[INFO] Building gitb-reports 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from maven central: https://repo1.maven.org/maven2/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.pom
[WARNING] The POM for com.lowagie:itext:jar:2.1.7.js10 is missing, no dependency information available
Downloading from maven central: https://repo1.maven.org/maven2/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.055 s
[INFO] Finished at: 2022-10-27T09:17:37+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project gitb-reports: Could not resolve dependencies for project com.gitb:gitb-reports:jar:1.0-SNAPSHOT: Could not find artifact com.lowagie:itext:jar:2.1.7.js10 in maven central (https://repo1.maven.org/maven2/) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
`
Regards,
Maurizio Longari

**Comment by costas80** [comment]
Hi @longama . Thanks for bringing this to our attention.

The library in question is a transitive dependency from jasperreports (the library used to generate PDF reports). Checking the library's [dependencies on Maven](https://mvnrepository.com/artifact/net.sf.jasperreports/jasperreports/6.20.0) I see that version `2.1.7.js10` is indeed missing from Maven Central with no information for an alternative repository. Oddly enough this seems to be a recent change as our builds have not failed earlier.

I will investigate further and come back to you with a fix (or at least a workaround) asap.

**Comment by costas80** [comment]
Ok I found the cause of the problem and why it fails for you but not our builds.

The missing artifact is located at https://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/, which in the project's `pom.xml` is defined as http://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/ (note `http` instead of `https`). Maven normally rejects dependencies to be pulled from non https repos unless you explicitly allow it in your Maven settings. To do this you would need to add in the mirrors section:
```
  <mirrors>
    <mirror>
      <id>jaspersoft-third-party-mirror</id>
      <mirrorOf>jaspersoft-third-party</mirrorOf>
      <url>http://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/</url>
      <blocked>false</blocked>
    </mirror>
  </mirrors>
```

Having said this, I see that Jaspersoft has just provided access to their JFrog repo over https as well meaning that all this would no longer be needed (this is a recent change, not available when we setup this dependency and repository reference). We will promptly update the pom.xml in the master branch to resolve this (I'll share an update here when this is done).

In the meantime, so that you aren't blocked, you can use the mirror workaround I mentioned above.

**Comment by costas80** [comment]
Hi @longama . The project's `pom.xml` is now updated to switch to the https version of the Jasper repo. This will resolve the build issue you faced.

Can you confirm that this fixes your problem?

**Comment by longama** [comment]
Hi @costas80 unfortunately this fix is not working on my system, I am getting the same error. I have tried to add an exclusion of itext library from jasper library in gitb-reports/pom.xml  this way:

`
        <dependency>
            <groupId>net.sf.jasperreports</groupId>
            <artifactId>jasperreports</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>com.lowagie</groupId>
                    <artifactId>itext</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
`
and it is building.

**Comment by costas80** [comment]
Hi @longama . I cannot reproduce the issue unfortunately. I tried a clean maven  build on a new VM (i.e. no existing cached artefacts or specific settings) and the the build is working without problems. On your own system you could try to do a containerised build so that your workstation's maven cache and settings are not involved. To do this issue `docker-compose build` from the repo's root folder.

Issuing this, all images and tools will be pulled, before proceeding with the maven build. Trying this out on my side I see the following output (relevant to the library in question):
```
[INFO] -----------------------< com.gitb:gitb-reports >------------------------
[INFO] Building gitb-reports 1.0-SNAPSHOT                                 [6/8]
[INFO] --------------------------------[ jar ]---------------------------------
...
Downloading from jaspersoft-third-party: https://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.pom
Downloaded from jaspersoft-third-party: https://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.pom (3.2 kB at 2.0 kB/s)
...
Downloading from jaspersoft-third-party: https://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.jar
Downloaded from jaspersoft-third-party: https://jaspersoft.jfrog.io/jaspersoft/third-party-ce-artifacts/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.jar (1.1 MB at 1.2 MB/s)
...
```

Checking again your initial comment, I see that only maven central is being queried for the library. This can be explained by maven altogether skipping http-based repositories. However, when you switch to https (as the latest POM does) things should work fine.

I think that following your initial maven build your local maven repository cache recorded the specific POM as missing and is not re-attempting to download it. To be certain, a simple thing to try would be to navigate to the resource's location in your maven repo (typically `HOME/.m2/repository/com/lowagie/itext/2.1.7.js10`) and just delete the `2.1.7.js10` folder. Doing so would force maven to re-download the resource using the https repository.

Finally, regarding your update to exclude the dependency as a whole, indeed this would result in the build succeeding. However, if you then run the Test Bed you will be getting ClassNotFound exceptions when making PDF exports (which is where this library comes in).

Could you try the fix I suggest (delete the cached library from your local repo) and let me know if the build is working?

**Comment by longama** [comment]
Hi @costas80 I followed your suggestion deleting the 2.1.7.js10 folder and rebuilt without success. I see the following logs in the build. It seems that it is not recognizing the repository specification. I tried wuth mvn 3.6.3 and  3.8.6.
``
[INFO] -----------------------< com.gitb:gitb-reports >------------------------
[INFO] Building gitb-reports 1.0-SNAPSHOT                                 [6/8]
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from maven central: https://repo1.maven.org/maven2/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.pom
[WARNING] The POM for com.lowagie:itext:jar:2.1.7.js10 is missing, no dependency information available
Downloading from maven central: https://repo1.maven.org/maven2/com/lowagie/itext/2.1.7.js10/itext-2.1.7.js10.jar
[INFO] ------------------------------------------------------------------------
``
